app/domain/authentication/authn_jwt/jwt_fetch_provider_uri_signing_key.rb

```diff
@@ -0,0 +1,55 @@
+module Authentication
+  module AuthnJwt
+
+    JwtFetchProviderUriSigningKey = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger,
+        discover_identity_provider: OAuth::DiscoverIdentityProvider.new,
+        fetch_secrets: Conjur::FetchRequiredSecrets.new
+      },
+      inputs: %i[authenticator_input]
+    ) do
+      def call
+        provider_uri
+        discover_provider
+        fetch_provider_keys
+      end
+
+      private
+
+      def provider_uri
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ProviderUriResourceNameConfiguration.new(resource_id))
+        @provider_uri_secret ||= @fetch_secrets.(resource_ids: [resource_id])
+        @provider_uri ||= @provider_uri_secret[resource_id]
+      end
+
+      def discover_provider
+        discovered_provider
+      end
+
+      def discovered_provider
+        @discovered_provider ||= @discover_identity_provider.(
+          provider_uri: @provider_uri
+        )
+      end
+
+      def fetch_provider_keys
+        jwks = {
+          keys: @discovered_provider.jwks
```

Comment:
please use the memoizing methods in all the PR (e.g `discovered_provider` instead of `@discovered_provider`